File: Archiver/app_code/clsEXCLUDEFROM.cs
using global::System.Data.SqlClient;
using System.Windows.Forms;
using global::Microsoft.VisualBasic;
using MODI;

namespace EcmArchiver
{
    public class clsEXCLUDEFROM
    {


        // ** DIM the selected table columns 
        private clsDatabaseARCH DBARCH = new clsDatabaseARCH();
        private clsDma DMA = new clsDma();
        private clsUtility UTIL = new clsUtility();
        private clsLogging LOG = new clsLogging();
        private string FromEmailAddr = "";
        private string SenderName = "";
        private string UserID = "";




        // ** Generate the SET methods 
        public void setFromemailaddr(ref string val)
        {
            if (Strings.Len(val) == 0)
            {
                MessageBox.Show("SET: Field 'Fromemailaddr' cannot be NULL.");
                return;
            }

            val = UTIL.RemoveSingleQuotes(val);
            FromEmailAddr = val;
        }

        public void setSendername(ref string val)
        {
            if (Strings.Len(val) == 0)
            {
                MessageBox.Show("SET: Field 'Sendername' cannot be NULL.");
                return;
            }

            val = UTIL.RemoveSingleQuotes(val);
            SenderName = val;
        }

        public void setUserid(ref string val)
        {
            if (Strings.Len(val) == 0)
            {
                MessageBox.Show("SET: Field 'Userid' cannot be NULL.");
                return;
            }

            val = UTIL.RemoveSingleQuotes(val);
            UserID = val;
        }






        // ** Generate the GET methods 
        public string getFromemailaddr()
        {
            if (Strings.Len(FromEmailAddr) == 0)
            {
                MessageBox.Show("GET: Field 'Fromemailaddr' cannot be NULL.");
                return "";
            }

            return UTIL.RemoveSingleQuotes(FromEmailAddr);
        }

        public string getSendername()
        {
            if (Strings.Len(SenderName) == 0)
            {
                MessageBox.Show("GET: Field 'Sendername' cannot be NULL.");
                return "";
            }

            return UTIL.RemoveSingleQuotes(SenderName);
        }

        public string getUserid()
        {
            if (Strings.Len(UserID) == 0)
            {
                MessageBox.Show("GET: Field 'Userid' cannot be NULL.");
                return "";
            }

            return UTIL.RemoveSingleQuotes(UserID);
        }






        // ** Generate the Required Fields Validation method 
        public bool ValidateReqData()
        {
            if (FromEmailAddr.Length == 0)
                return false;
            if (SenderName.Length == 0)
                return false;
            if (UserID.Length == 0)
                return false;
            return true;
        }




        // ** Generate the Validation method 
        public bool ValidateData()
        {
            if (FromEmailAddr.Length == 0)
                return false;
            if (SenderName.Length == 0)
                return false;
            if (UserID.Length == 0)
                return false;
            return true;
        }




        // ** Generate the INSERT method 
        public bool Insert()
        {
            bool b = false;
            string s = "";
            s = s + " INSERT INTO ExcludeFrom(";
            s = s + "FromEmailAddr,";
            s = s + "SenderName,";
            s = s + "UserID) values (";
            s = s + "'" + FromEmailAddr + "'" + ",";
            s = s + "'" + SenderName + "'" + ",";
            s = s + "'" + UserID + "'" + ")";
            return DBARCH.ExecuteSqlNewConn(s, false);
        }




        // ** Generate the UPDATE method 
        public bool Update(string WhereClause)
        {
            bool b = false;
            string s = "";
            if (Strings.Len(WhereClause) == 0)
                return false;
            s = s + " update ExcludeFrom set ";
            s = s + "FromEmailAddr = '" + getFromemailaddr() + "'" + ", ";
            s = s + "SenderName = '" + getSendername() + "'" + ", ";
            s = s + "UserID = '" + getUserid() + "'";
            WhereClause = " " + WhereClause;
            s = s + WhereClause;
            return DBARCH.ExecuteSqlNewConn(s, false);
        }




        // ** Generate the SELECT method 
        public SqlDataReader SelectRecs()
        {
            bool b = false;
            string s = "";
            var rsData = default(SqlDataReader);
            s = s + " SELECT ";
            s = s + "FromEmailAddr,";
            s = s + "SenderName,";
            s = s + "UserID ";
            s = s + " FROM ExcludeFrom";
            string CS = DBARCH.setConnStr();     // DBARCH.getGateWayConnStr(gGateWayID) : Dim CONN As New SqlConnection(CS) : CONN.Open() : Dim command As New SqlCommand(s, CONN) : rsData = command.ExecuteReader()
            return rsData;
        }




        // ** Generate the Select One Row method 
        public SqlDataReader SelectOne(string WhereClause)
        {
            bool b = false;
            string s = "";
            var rsData = default(SqlDataReader);
            s = s + " SELECT ";
            s = s + "FromEmailAddr,";
            s = s + "SenderName,";
            s = s + "UserID ";
            s = s + " FROM ExcludeFrom";
            s = s + WhereClause;
            string CS = DBARCH.setConnStr();     // DBARCH.getGateWayConnStr(gGateWayID) : Dim CONN As New SqlConnection(CS) : CONN.Open() : Dim command As New SqlCommand(s, CONN) : rsData = command.ExecuteReader()
            return rsData;
        }




        // ** Generate the DELETE method 
        public bool Delete(string WhereClause)
        {
            bool b = false;
            string s = "";
            if (Strings.Len(WhereClause) == 0)
                return false;
            WhereClause = " " + WhereClause;
            s = " Delete from ExcludeFrom";
            s = s + WhereClause;
            b = DBARCH.ExecuteSqlNewConn(s, false);
            return b;
        }




        // ** Generate the Zeroize Table method 
        public bool Zeroize()
        {
            bool b = false;
            string s = "";
            s = s + " Delete from ExcludeFrom";
            b = DBARCH.ExecuteSqlNewConn(s, false);
            return b;
        }




        // ** Generate Index Queries 
        public int cnt_Pi01_ExcludeFrom(string FromEmailAddr, string UserID)
        {
            int B = 0;
            string TBL = "ExcludeFrom";
            string WC = "Where FromEmailAddr = '" + FromEmailAddr + "' and   UserID = '" + UserID + "'";
            B = DBARCH.iGetRowCount(TBL, WC);
            return B;
        }     // ** cnt_Pi01_ExcludeFrom

        public int cnt_PK_ExcludeFrom(string FromEmailAddr, string SenderName, string UserID)
        {
            int B = 0;
            string TBL = "ExcludeFrom";
            string WC = "Where FromEmailAddr = '" + FromEmailAddr + "' and   SenderName = '" + SenderName + "' and   UserID = '" + UserID + "'";
            B = DBARCH.iGetRowCount(TBL, WC);
            return B;
        }     // ** cnt_PK_ExcludeFrom


        // ** Generate Index ROW Queries 
        public SqlDataReader getRow_Pi01_ExcludeFrom(string FromEmailAddr, string UserID)
        {
            SqlDataReader rsData = null;
            string TBL = "ExcludeFrom";
            string WC = "Where FromEmailAddr = '" + FromEmailAddr + "' and   UserID = '" + UserID + "'";
            rsData = DBARCH.GetRowByKey(TBL, WC);
            if (rsData.HasRows)
            {
                return rsData;
            }
            else
            {
                return null;
            }
        }     // ** getRow_Pi01_ExcludeFrom

        public SqlDataReader getRow_PK_ExcludeFrom(string FromEmailAddr, string SenderName, string UserID)
        {
            SqlDataReader rsData = null;
            string TBL = "ExcludeFrom";
            string WC = "Where FromEmailAddr = '" + FromEmailAddr + "' and   SenderName = '" + SenderName + "' and   UserID = '" + UserID + "'";
            rsData = DBARCH.GetRowByKey(TBL, WC);
            if (rsData.HasRows)
            {
                return rsData;
            }
            else
            {
                return null;
            }
        }     // ** getRow_PK_ExcludeFrom


        /// Build Index Where Caluses
        public string wc_Pi01_ExcludeFrom(string FromEmailAddr, string UserID)
        {
            string WC = "Where FromEmailAddr = '" + FromEmailAddr + "' and   UserID = '" + UserID + "'";
            return WC;
        }     // ** wc_Pi01_ExcludeFrom

        public string wc_PK_ExcludeFrom(string FromEmailAddr, string SenderName, string UserID)
        {
            string WC = "Where FromEmailAddr = '" + FromEmailAddr + "' and   SenderName = '" + SenderName + "' and   UserID = '" + UserID + "'";
            return WC;
        }     // ** wc_PK_ExcludeFrom
    }
}

File: Archiver/app_code/clsEXCLUDEFROM.vb
Imports Microsoft.VisualBasic
Imports System.Configuration
Imports System
Imports System.Data.SqlClient
Imports System.Collections
Imports System.IO
Imports System.Data.Sql


Public Class clsEXCLUDEFROM


    '** DIM the selected table columns 
    Dim DBARCH As New clsDatabaseARCH
    Dim DMA As New clsDma
    Dim UTIL As New clsUtility
    Dim LOG As New clsLogging



    Dim FromEmailAddr As String = ""
    Dim SenderName As String = ""
    Dim UserID As String = ""




    '** Generate the SET methods 
    Public Sub setFromemailaddr(ByRef val As String)
        If Len(val) = 0 Then
            MessageBox.Show("SET: Field 'Fromemailaddr' cannot be NULL.")
            Return
        End If
        val = UTIL.RemoveSingleQuotes(val)
        FromEmailAddr = val
    End Sub


    Public Sub setSendername(ByRef val As String)
        If Len(val) = 0 Then
            MessageBox.Show("SET: Field 'Sendername' cannot be NULL.")
            Return
        End If
        val = UTIL.RemoveSingleQuotes(val)
        SenderName = val
    End Sub


    Public Sub setUserid(ByRef val As String)
        If Len(val) = 0 Then
            MessageBox.Show("SET: Field 'Userid' cannot be NULL.")
            Return
        End If
        val = UTIL.RemoveSingleQuotes(val)
        UserID = val
    End Sub






    '** Generate the GET methods 
    Public Function getFromemailaddr() As String
        If Len(FromEmailAddr) = 0 Then
            MessageBox.Show("GET: Field 'Fromemailaddr' cannot be NULL.")
            Return ""
        End If
        Return UTIL.RemoveSingleQuotes(FromEmailAddr)
    End Function


    Public Function getSendername() As String
        If Len(SenderName) = 0 Then
            MessageBox.Show("GET: Field 'Sendername' cannot be NULL.")
            Return ""
        End If
        Return UTIL.RemoveSingleQuotes(SenderName)
    End Function


    Public Function getUserid() As String
        If Len(UserID) = 0 Then
            MessageBox.Show("GET: Field 'Userid' cannot be NULL.")
            Return ""
        End If
        Return UTIL.RemoveSingleQuotes(UserID)
    End Function






    '** Generate the Required Fields Validation method 
    Public Function ValidateReqData() As Boolean
        If FromEmailAddr.Length = 0 Then Return False
        If SenderName.Length = 0 Then Return False
        If UserID.Length = 0 Then Return False
        Return True
    End Function




    '** Generate the Validation method 
    Public Function ValidateData() As Boolean
        If FromEmailAddr.Length = 0 Then Return False
        If SenderName.Length = 0 Then Return False
        If UserID.Length = 0 Then Return False
        Return True
    End Function




    '** Generate the INSERT method 
    Public Function Insert() As Boolean
        Dim b As Boolean = False
        Dim s As String = ""
        s = s + " INSERT INTO ExcludeFrom("
        s = s + "FromEmailAddr,"
        s = s + "SenderName,"
        s = s + "UserID) values ("
        s = s + "'" + FromEmailAddr + "'" + ","
        s = s + "'" + SenderName + "'" + ","
        s = s + "'" + UserID + "'" + ")"
        Return DBARCH.ExecuteSqlNewConn(s, False)


    End Function




    '** Generate the UPDATE method 
    Public Function Update(ByVal WhereClause As String) As Boolean
        Dim b As Boolean = False
        Dim s As String = ""


        If Len(WhereClause) = 0 Then Return False


        s = s + " update ExcludeFrom set "
        s = s + "FromEmailAddr = '" + getFromemailaddr() + "'" + ", "
        s = s + "SenderName = '" + getSendername() + "'" + ", "
        s = s + "UserID = '" + getUserid() + "'"
        WhereClause = " " + WhereClause
        s = s + WhereClause
        Return DBARCH.ExecuteSqlNewConn(s, False)
    End Function




    '** Generate the SELECT method 
    Public Function SelectRecs() As SqlDataReader
        Dim b As Boolean = False
        Dim s As String = ""
        Dim rsData As SqlDataReader
        s = s + " SELECT "
        s = s + "FromEmailAddr,"
        s = s + "SenderName,"
        s = s + "UserID "
        s = s + " FROM ExcludeFrom"

        Dim CS As String = DBARCH.setConnStr()     'DBARCH.getGateWayConnStr(gGateWayID) : Dim CONN As New SqlConnection(CS) : CONN.Open() : Dim command As New SqlCommand(s, CONN) : rsData = command.ExecuteReader()
        Return rsData
    End Function




    '** Generate the Select One Row method 
    Public Function SelectOne(ByVal WhereClause As String) As SqlDataReader
        Dim b As Boolean = False
        Dim s As String = ""
        Dim rsData As SqlDataReader
        s = s + " SELECT "
        s = s + "FromEmailAddr,"
        s = s + "SenderName,"
        s = s + "UserID "
        s = s + " FROM ExcludeFrom"
        s = s + WhereClause

        Dim CS As String = DBARCH.setConnStr()     'DBARCH.getGateWayConnStr(gGateWayID) : Dim CONN As New SqlConnection(CS) : CONN.Open() : Dim command As New SqlCommand(s, CONN) : rsData = command.ExecuteReader()
        Return rsData
    End Function




    '** Generate the DELETE method 
    Public Function Delete(ByVal WhereClause As String) As Boolean
        Dim b As Boolean = False
        Dim s As String = ""


        If Len(WhereClause) = 0 Then Return False


        WhereClause = " " + WhereClause


        s = " Delete from ExcludeFrom"
        s = s + WhereClause


        b = DBARCH.ExecuteSqlNewConn(s, False)
        Return b


    End Function




    '** Generate the Zeroize Table method 
    Public Function Zeroize() As Boolean
        Dim b As Boolean = False
        Dim s As String = ""


        s = s + " Delete from ExcludeFrom"


        b = DBARCH.ExecuteSqlNewConn(s, False)
        Return b


    End Function




    '** Generate Index Queries 
    Public Function cnt_Pi01_ExcludeFrom(ByVal FromEmailAddr As String, ByVal UserID As String) As Integer


        Dim B As Integer = 0
        Dim TBL As String = "ExcludeFrom"
        Dim WC As String = "Where FromEmailAddr = '" + FromEmailAddr + "' and   UserID = '" + UserID + "'"


        B = DBARCH.iGetRowCount(TBL, WC)


        Return B
    End Function     '** cnt_Pi01_ExcludeFrom
    Public Function cnt_PK_ExcludeFrom(ByVal FromEmailAddr As String, ByVal SenderName As String, ByVal UserID As String) As Integer


        Dim B As Integer = 0
        Dim TBL As String = "ExcludeFrom"
        Dim WC As String = "Where FromEmailAddr = '" + FromEmailAddr + "' and   SenderName = '" + SenderName + "' and   UserID = '" + UserID + "'"


        B = DBARCH.iGetRowCount(TBL, WC)


        Return B
    End Function     '** cnt_PK_ExcludeFrom


    '** Generate Index ROW Queries 
    Public Function getRow_Pi01_ExcludeFrom(ByVal FromEmailAddr As String, ByVal UserID As String) As SqlDataReader


        Dim rsData As SqlDataReader = Nothing
        Dim TBL As String = "ExcludeFrom"
        Dim WC As String = "Where FromEmailAddr = '" + FromEmailAddr + "' and   UserID = '" + UserID + "'"


        rsData = DBARCH.GetRowByKey(TBL, WC)


        If rsData.HasRows Then
            Return rsData
        Else
            Return Nothing
        End If
    End Function     '** getRow_Pi01_ExcludeFrom
    Public Function getRow_PK_ExcludeFrom(ByVal FromEmailAddr As String, ByVal SenderName As String, ByVal UserID As String) As SqlDataReader


        Dim rsData As SqlDataReader = Nothing
        Dim TBL As String = "ExcludeFrom"
        Dim WC As String = "Where FromEmailAddr = '" + FromEmailAddr + "' and   SenderName = '" + SenderName + "' and   UserID = '" + UserID + "'"


        rsData = DBARCH.GetRowByKey(TBL, WC)


        If rsData.HasRows Then
            Return rsData
        Else
            Return Nothing
        End If
    End Function     '** getRow_PK_ExcludeFrom


    ''' Build Index Where Caluses 
    '''
    Public Function wc_Pi01_ExcludeFrom(ByVal FromEmailAddr As String, ByVal UserID As String) As String


Dim WC AS String  = "Where FromEmailAddr = '" + FromEmailAddr + "' and   UserID = '" + UserID + "'" 


        Return WC
    End Function     '** wc_Pi01_ExcludeFrom
    Public Function wc_PK_ExcludeFrom(ByVal FromEmailAddr As String, ByVal SenderName As String, ByVal UserID As String) As String


Dim WC AS String  = "Where FromEmailAddr = '" + FromEmailAddr + "' and   SenderName = '" + SenderName + "' and   UserID = '" + UserID + "'" 


        Return WC
    End Function     '** wc_PK_ExcludeFrom
End Class
